_posts/2023-03-06-NGINX-Dashboards-with-Fluent-Bit.md

```diff
@@ -0,0 +1,68 @@
+title:  "Simple NGINX Dashboards with Fluent Bit and OpenSearch"
+authors:
+  - jdbright
+  - lioperry
+  - anurag_gup
+  - armarble
+date:   2023-03-06
+categories:
+  - technical-post
+meta_keywords: opensearch simple schema, observability schema, opensearch opentelemetry, OpenSearch 2.6
+meta_description: While users can send their logs to OpenSearch today, there are no set formats or schemas which can make sharing dashboards and alerts cumbersome. In this blog, we talk about using Fluent Bit, a new simple schema and OpenSearch with Nginx as the first example.
+---
+
+Fluent Bit is a graduated sub-project under the Cloud Native Computing Foundation (CNCF) Fluentd project umbrella. Fluent Bit has hundreds of integrations to common tools such as Kafka, Syslog, Loki, and of course OpenSearch.
+
+While users can send their logs to OpenSearch today, there are no set formats or schemas, which can make sharing dashboards and alerts cumbersome. In this blog, we talk about using Fluent Bit, a new simple schema, and OpenSearch with Nginx, as the first example.
+
+## Simple Schema
+
+OpenSearch 2.6 introduced a standardization for conforming to a common and unified observability schema: Simple Schema for Observability. 
+
+Observability is a collection of plugins and applications that let you visualize data-driven events by using PPL / SQL / DQL to explore and query data stored in OpenSearch.
+With the schema in place, Observability tools can ingest, automatically extract, and aggregate data and create custom dashboards, making it easier to understand the system at a higher level.
+
+Simple Schema for Observability is inspired by both [OpenTelemetry](https://opentelemetry.io/docs/) and uses Amazon Elastic Container Service ([Amazon ECS](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs_cwe_events.html)) event logs and OpenTelemetry metadata.
+
+## Connecting Simple Schema and Fluent Bit
+
+When raw logs are ingested by Fluent Bit they are automatically converted into MessagePack, a binary representation of JSON. Fluent Bit’s automatic conversion makes parsing and modifying messages in-flight simple and easy.
+
+In addition, Fluent Bit comes with out of the box parsers for common applications such as Nginx, Apache Web Logs, Kubernetes, and more. These parsers give structure to a log file, though they do not have the depth that other log schemas and formats do. Thankfully, Fluent Bit can also handle more complex transformations along with with processors (also known as filters), or even Lua code.
+
+[Lua filters](https://docs.fluentbit.io/manual/pipeline/filters/lua) give users extreme flexibility with how they want to transform their data, including modifying, adding, enriching via API, calculating, or even redacting. To help us connect Fluent Bit and the new simple schema, we opted for a Lua script that you can paste into your config file to run on top of NGINX ingested logs.
+
+fluent-bit.conf example for VM or standalone deployments:
```

Comment:
```suggestion
The following is a fluent-bit.conf example for VM or standalone deployments:
```

Reply:
Do we mean "The following is an example fluent-bit.conf [noun] for..."? Does VM stand for "virtual machine" here?